io/pci/pci_hotplug.py

```diff
@@ -131,8 +114,31 @@ def hotplug_add(self, slot, pci_addr):
         genio.write_file("/sys/bus/pci/slots/%s/power" % slot, "1")
 
         def is_added():
+            """
+            Returns True if pci device is added, False otherwise.
+            """
             if pci_addr not in pci.get_pci_addresses():
                 return False
             return True
 
-        return wait.wait_for(is_added, timeout=10) or False
+        def is_recovered():
+            """
+            Compares current endpoint devices in pci address with
+            `pre` value, and returns True if equals pre.
+            False otherwise.
+            """
+            post = len(pci.get_disks_in_pci_address(pci_addr))
+            post += len(pci.get_nics_in_pci_address(pci_addr))
+            if post == self.end_devices[pci_addr]:
+                self.log.debug("Pre: %d,  Post: %d",
```

Comment:
Can we move this message before checking since it is anyway printed?

Reply:
Thanks, taken care of.